videoframegrid.py

```diff
@@ -68,6 +71,28 @@ def swap_frame(self, frame):
             else:
                 self.selected_frame = frame
 
+    def toggle_fullscreen(self, selected_frame):
+        if not self.fullscreen:
```

Comment:
Perhaps `if self.fullscreen` and do the else first? Flavor thing really your choice